### PR TITLE
macvim: remove luajit.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
See comments in https://github.com/Homebrew/homebrew-core/pull/34411#issuecomment-442127740

Fixes https://github.com/homebrew/homebrew-core/issues/34527

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Maintainers: feel free to merge this when ✅ as I'm going on vacation soon.

**Review by fxcoudert — APPROVED** [review]

**Pull request closed without merging**
